test/test_readonly_task.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+require "syskit/test/self"
+
+module Syskit
+    describe TaskContext do
+        before do
+            task_m = TaskContext.new_submodel do
+                property "p", "/double"
+            end
+
+            @task = syskit_stub_deploy_and_configure(task_m.with_arguments(read_only: true))
+            Orocos.allow_blocking_calls { @task.orocos_task.configure(false) }
+        end
+
+        after do
+            state = Orocos.allow_blocking_calls { @task.orocos_task.rtt_state }
+            if state == :RUNNING
+                Orocos.allow_blocking_calls { @task.orocos_task.stop(false) }
+            end
+
+            expect_execution { @task.stop! } if @task.started?
+        end
+
+        describe "read-only tasks" do
+            it "raises when attempting to change a property" do
+                task = @task
+
+                assert_raises(InvalidReadOnlyOperation) do
+                    task.properties.p = 2.0
+                end
+            end
+
+            it "emits start when it is started while the component is running" do
+                task = @task
+
+                expect_execution do
+                    Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+                    task.start!
+                end.to { emit task.start_event }
+            end
+
+            it "does not emit start if started while the component is not running" do
+                task = @task
+
+                expect_execution do
+                    Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+                end.to { not_emit task.start_event }
+            end
+
+            it "emits start if it is already running and the component itself is started" do
+                task = @task
+
+                # starting the component {state will be :RUNNING}
+                Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+                expect_execution { task.start! }
+                    .to { emit task.start_event }
+
+                assert task.started?
+
+                expect_execution { task.start! }
+                    .to { emit task.start_event }
+            end
+
+            it "never stop the component" do
+                task = @task
+
+                # here it's already stopped
+                expect_execution { task.stop! }
+                    .to { emit task.stop_event }
+
+                # starting the component {state will be :RUNNING}
+                Orocos.allow_blocking_calls { task.orocos_task.start(false) }
+                expect_execution { task.start! }
```

Comment:
Here the task fails to start due to: 
> Roby::TaskEventNotExecutable: start_event.emit on :0x5653221596d8<id:21>(conf: ["default"], orocos_name: stub3, read_only: true) which is not executable

For some reason, `task.executable?` is nil after explicitly calling `stop!`

Reply:
This might be because you can't restart a Syskit task.